src/transactions/api.ts

```diff
@@ -0,0 +1,33 @@
+import { createApi } from '@reduxjs/toolkit/query/react'
+import { baseQuery, isRehydrateAction } from 'src/redux/api'
+import type { TokenTransaction } from 'src/transactions/types'
+
+export type TransactionFeedV2Response = {
+  transactions: TokenTransaction[]
+  pageInfo: {
+    hasNextPage: boolean
+  }
+}
+
+export const transactionFeedV2Api = createApi({
+  reducerPath: 'transactionFeedV2Api',
+  baseQuery,
+  endpoints: (builder) => ({
+    transactionFeedV2: builder.query<
+      TransactionFeedV2Response,
+      { address: string; endCursor: number }
+    >({
+      query: ({ address, endCursor }) => `/wallet/${address}/transactions?endCursor=${endCursor}`,
+    }),
+  }),
+  extractRehydrationInfo: (action, { reducerPath }): any => {
+    if (isRehydrateAction(action)) {
+      /**
+       * Even though payload is types as RootState - redux-persist can evaluate it as undefined.
+       */
+      return action.payload?.[reducerPath]
```

Comment:
So this means we'll persist and restore the API slice, right?

If that's correct, how we'll we handle changes to the API response that would require a client side migration?

Reply:
@jeanregisser Sorry, I will remove the whole `extractRehydrationInfo` part!

When I initially added RTK-Query, I was trying a lot of variations to try to keep api persisted so I've added this handler but forgot to remove it once it was no longer needed.